cmd/scheduled-feed/main.go

```diff
@@ -26,17 +27,50 @@ type FeedHandler struct {
 }
 
 func (handler *FeedHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var err error
+	pkgs, pollErrors := handler.pollFeeds()
+	processedPackages, err := handler.publishPackages(pkgs)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if pollErrors {
+		http.Error(w, "error polling for packages - see logs for more information", http.StatusInternalServerError)
+		return
+	}
+	_, err = w.Write([]byte(fmt.Sprintf("%d packages processed", processedPackages)))
+	if err != nil {
+		http.Error(w, "unexpected error during http server write: %w", http.StatusInternalServerError)
+	}
+}
+
+func (handler FeedHandler) getCutoff() time.Time {
+	var cutoff time.Time
+	if handler.lastPoll.IsZero() {
+		cutoff = time.Now().UTC().Add(-handler.pollRate)
+	} else {
+		cutoff = handler.lastPoll
+	}
+	return cutoff
+}
+
+func (handler *FeedHandler) pollFeeds() ([]*feeds.Package, bool) {
 	cutoff := handler.getCutoff()
 	handler.lastPoll = time.Now().UTC()
 	pkgs, errs := handler.scheduler.Poll(cutoff)
+	errors := false
 	if len(errs) > 0 {
+		errors = true
 		for _, err := range errs {
 			log.Errorf("error polling for new packages: %v", err)
```

Comment:
Whilst this isn't part of this PR, this seems duplicative as the scheduler already has `logger.WithError(result.err).Error("error fetching packages")` as it appends to the errors return value.

Reply:
Yep, thanks for raising https://github.com/ossf/package-feeds/issues/100